Audio/BnkCompiler/Compiler.cs

```diff
@@ -16,24 +19,282 @@ public class CompileResult
         public CompilerData Project { get; set; }
         public PackFile OutputBnkFile { get; set; }
         public PackFile OutputDatFile { get; set; }
-        public PackFile NameList { get; set; }
+        public PackFile OutputStatesDatFile{ get; set; }
     }
 
     internal static class CompilerConstants
     {
         public static readonly string Game_Warhammer3 = "Warhammer3";
-        public static readonly string Language_English = "English(uk)";
-        public static readonly string Mixers_Diplomacy = "54848735";
+        public static readonly Dictionary<string, uint> VanillaIds = new Dictionary<string, uint>()
+        {
+            // Ids for Event mixers
+
+            {"advisor", 517250292},
+            {"battle individual vocalisation", 508226369},
+            {"diplomacy line", 54848735},
+            {"ind magic", 140075115},
+            {"movie", 573597124},
+            {"quest battle", 659413513},
+            {"ui", 608071769},
+            {"event narration", 517250292},
+
+            // Ids for Dialogue Event mixers 
+            {"battle_vo_conversational", 600762068},
+            {"battle_vo_orders", 1009314120},
+            {"campaign_vo_conversational", 652491101},
+            {"campaign_vo", 306196174},
+            {"frontend_vo", 745637913},
+
+            // Ids for attenuation objects for Dialogue Events 
+            {"battle_vo_conversational_attenuation", 649943956},
+            {"battle_vo_orders_attenuation", 803409642},
+            {"campaign_vo_conversational_attenuation", 62329658},
+            {"campaign_vo_attenuation", 432982952},
+        };
+
+        public static string MatchDialogueEventToBnk(string dialogueEvent)
+        {
+            dialogueEvent = dialogueEvent.ToLower();
+
+            if (dialogueEvent.Contains("battle_vo_conversation"))
+                return "battle_vo_conversational";
+            else if (dialogueEvent.Contains("battle_vo_order"))
+                return "battle_vo_orders";
+            else if (dialogueEvent.Contains("campaign_vo_cs"))
+                return "campaign_vo_conversational";
+            else if (dialogueEvent.Contains("campaign_vo") || dialogueEvent == "gotrek_felix_arrival" || dialogueEvent == "gotrek_felix_departure")
+                return "campaign_vo";
+            else if (dialogueEvent.Contains("frontend_vo"))
+                return "frontend_vo";
+            else if (dialogueEvent == "battle_individual_melee_weapon_hit")
+                return "battle_individual_melee";
+            else
+                throw new Exception("Dialogue_Event could not be matched to a bnk.");
+        }
 
-        public static readonly string[] Possible_Languages = new string[] { Language_English };
+        public static Dictionary<string, List<string>> dialogueEventStates = new Dictionary<string, List<string>>()
```

Comment:
Large lists of constants does not really belong alongside business logic. 
I suggest moving it to a different class which implements an interface.The class should be inside the warhammer 3 specific folder. 
As an alternative, consider budling this dataset dynamically as it can (and probably will) change with every patch and dlc

Reply:
Cool will look into some way to extract this dynamically either by the event_data__core.dat or by just reading all dialogue events and getting their arguments list.